packages/generator/src/migrate.ts

```diff
@@ -258,63 +325,169 @@ function getColumnType(details: SearchParameterDetails): string {
   b.newLine();
-  b.append('await client.query(`CREATE TABLE IF NOT EXISTS "' + tableName + '" (');
+  b.append('export async function run(client: PoolClient): Promise<void> {');
   b.indentCount++;
-  b.append('"id" UUID NOT NULL PRIMARY KEY,');
-  b.append('"resourceId" UUID NOT NULL,');
-  b.append('"index" INTEGER NOT NULL,');
-  b.append('"content" TEXT NOT NULL,');
-  for (let i = 0; i < columns.length; i++) {
-    b.append(`"${columns[i]}" TEXT` + (i !== columns.length - 1 ? ',' : ''));
+
+  for (const targetTable of targetDefinition.tables) {
+    const startTable = startDefinition.tables.find((t) => t.name === targetTable.name);
+    migrateTable(b, startTable, targetTable);
   }
+
   b.indentCount--;
-  b.append(')`);');
-  b.newLine();
-  for (const column of columns) {
-    b.append(`await client.query('CREATE INDEX ON "${tableName}" ("${column}")');`);
+  b.append('}');
+}
+
+function migrateTable(b: FileBuilder, startTable: TableDefinition | undefined, targetTable: TableDefinition): void {
+  if (!startTable) {
+    writeCreateTable(b, targetTable);
+  } else {
+    migrateColumns(b, startTable, targetTable);
+    migrateIndexes(b, startTable, targetTable);
   }
 }
 
-function buildValueSetElementTable(b: FileBuilder): void {
+function writeCreateTable(b: FileBuilder, tableDefinition: TableDefinition): void {
   b.newLine();
-  b.append('await client.query(`CREATE TABLE IF NOT EXISTS "ValueSetElement" (');
+  b.appendNoWrap(`await client.query(\`CREATE TABLE IF NOT EXISTS "${tableDefinition.name}" (`);
   b.indentCount++;
-  b.append('"id" UUID NOT NULL PRIMARY KEY,');
-  b.append('"system" TEXT,');
-  b.append('"code" TEXT,');
-  b.append('"display" TEXT');
+  for (let i = 0; i < tableDefinition.columns.length; i++) {
+    b.append(
+      `"${tableDefinition.columns[i].name}" ${tableDefinition.columns[i].type}` +
+        (i !== tableDefinition.columns.length - 1 ? ',' : '')
+    );
+  }
   b.indentCount--;
-  b.append(')`);');
+  b.append('`);');
+  b.newLine();
+
+  for (const indexDefinition of tableDefinition.indexes) {
+    b.appendNoWrap(`await client.query('${buildIndexSql(tableDefinition.name, indexDefinition)}');`);
+  }
   b.newLine();
-  b.append(`await client.query('CREATE INDEX ON "ValueSetElement" ("system")');`);
-  b.append(`await client.query('CREATE INDEX ON "ValueSetElement" ("code")');`);
-  b.append(`await client.query('CREATE INDEX ON "ValueSetElement" ("display")');`);
+}
+
+function migrateColumns(b: FileBuilder, startTable: TableDefinition, targetTable: TableDefinition): void {
+  for (const targetColumn of targetTable.columns) {
+    const startColumn = startTable.columns.find((c) => c.name === targetColumn.name);
+    if (!startColumn) {
+      writeAddColumn(b, targetTable, targetColumn);
+    } else if (
+      startColumn.type.replaceAll('TIMESTAMP WITH TIME ZONE', 'TIMESTAMPTZ').trim() !==
+      targetColumn.type.replaceAll(' PRIMARY KEY', '').replaceAll(' DEFAULT FALSE', '').trim()
```

Comment:
What's going on here?

Reply:
Yeah...  This is a mismatch between "column definition" and "index definition"

I agree with your meta point, which is "this is confusing".... I'll move this logic to a helper function called "normalizeColumnDefinition" or something.